src/storage/conversion/ContainerToTemplateConverter.ts

```diff
@@ -0,0 +1,110 @@
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { INTERNAL_QUADS } from '../../util/ContentTypes';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import type { IdentifierStrategy } from '../../util/identifiers/IdentifierStrategy';
+import { isContainerIdentifier, isContainerPath } from '../../util/PathUtil';
+import type { TemplateEngine } from '../../util/templates/TemplateEngine';
+import { LDP } from '../../util/Vocabularies';
+import type { RepresentationConverterArgs } from './RepresentationConverter';
+import { TypedRepresentationConverter } from './TypedRepresentationConverter';
+
+interface ResourceDetails {
+  name: string;
+  identifier: string;
+  container: boolean;
+}
+
+/**
+ * A {@link RepresentationConverter} that creates a templated representation of a container.
+ */
+export class ContainerToTemplateConverter extends TypedRepresentationConverter {
+  private readonly identifierStrategy: IdentifierStrategy;
+  private readonly templateEngine: TemplateEngine;
+  private readonly contentType: string;
+
+  public constructor(templateEngine: TemplateEngine, contentType: string, identifierStrategy: IdentifierStrategy) {
+    super(INTERNAL_QUADS, contentType);
+    this.templateEngine = templateEngine;
+    this.contentType = contentType;
+    this.identifierStrategy = identifierStrategy;
+  }
+
+  public async canHandle(args: RepresentationConverterArgs): Promise<void> {
+    if (!isContainerIdentifier(args.identifier)) {
+      throw new NotImplementedHttpError('Can only convert containers.');
+    }
+    await super.canHandle(args);
+  }
+
+  public async handle({ identifier, representation }: RepresentationConverterArgs): Promise<Representation> {
+    const rendered = await this.templateEngine.render({
+      identifier: identifier.path,
+      name: this.getLocalName(identifier.path),
+      container: true,
+      children: await this.getChildResources(identifier, representation.data),
+      parents: this.getParentContainers(identifier),
+    });
+    return new BasicRepresentation(rendered, representation.metadata, this.contentType);
+  }
+
+  /**
+   * Collects the children of the container as simple objects.
+   */
+  private async getChildResources(container: ResourceIdentifier, quads: Readable): Promise<ResourceDetails[]> {
+    // Collect the needed bits of information from the containment triples
+    const resources = new Set<string>();
+    quads.on('data', ({ subject, predicate, object }: Quad): void => {
+      if (subject.value === container.path && predicate.equals(LDP.terms.contains)) {
+        resources.add(object.value);
+      }
+    });
+
+    // Transform the collected bits into an array of objects
+    return new Promise((resolve, reject): void => {
+      quads.on('end', (): void => {
+        // Create an object for every resource
+        const children = new Array<ResourceDetails>();
```

Comment:
Does this have an advantage compared to `const children: ResourceDetails[] = [];`?